net/host.go

```diff
@@ -140,7 +140,7 @@ func NewHost(cfg *Config) (*host, error) {
 	}
 
 	// create libp2p host instance
-	h, err := libp2p.New(context.Background(), opts...)
```

Comment:
I guess the context wasn't getting used: https://github.com/libp2p/go-libp2p/commit/b7bee3855cb86e50440e23b463605ea874c38787